### PR TITLE
Fix nested paths

### LINKED ISSUE #144: Nested path not created for PutObject call
Expected PutObject(bucket,"/path/to/object") to succeed. Instead the / gets stripped and object with name pathtoobject is created.

### PR DESCRIPTION
**Pull request by poornas**
minio-dotnet erroneously constructs nested paths without / delimiter. Put object to nested path should succeed seamlessly. This PR fixes #144

### CODE REVIEW

File: Minio/Helper/utils.cs [hunk 1]
```diff
@@ -99,10 +100,49 @@ internal static void validateObjectPrefix(string objectPrefix)
...
+                                              .Replace("\\*", "%2A")
+                                              .Replace("\\+", "%2B")
+                                              .Replace("\\,", "%2C")
+                                              .Replace("\\/", "%2F")
+                                              .Replace("\\:", "%3A")
+                                              .Replace("\\;", "%3B")
+                                              .Replace("\\=", "%3D")
+                                              .Replace("\\@", "%40")
+                                              .Replace("\\[", "%5B")
+                                              .Replace("\\]", "%5D");
```
Comment (NitishT) [inline Minio/Helper/utils.cs]: can you add some comments here @poornas

File: Minio/MinioClient.cs [hunk 2]
```diff
@@ -185,11 +186,7 @@ private string FullUserAgent
             if (objectName != null)
             {
                 // Limitation: OkHttp does not allow to add '.' and '..' as path segment.
```
Comment (NitishT) [inline Minio/MinioClient.cs]: This comment is probably old, pls remove

Comment (harshavardhana) [inline Minio/MinioClient.cs]: Actually `okhttp` has no meaning here in dotnet its a java library @poornas

**Comment by poornas** [comment]
@NitishT, review comments fixed

**Comment by poornas** [comment]
@NitishT , awaiting your review.

**Comment by poornas** [comment]
@harshavardhana , can we merge this to master before the next release is published to nuget.

**Pull request merged**